### PR TITLE
Dev/20221001 init

### PR DESCRIPTION
**Pull request by mrk-andreev**

### CODE REVIEW

File: src/app_predictor/app_predictor.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+                or (rect.y + rect.h) > img.shape[0]
+                or rect.x < 0
+                or rect.x > img.shape[1]
+                or rect.x + rect.w > img.shape[1]
+        ):
+            warnings.warn("Rect out of image")
+            continue
+
+        img[rect.y:rect.y + rect.h, rect.x:rect.x + rect.w, :] = cv2.blur(
+            img[rect.y:rect.y + rect.h, rect.x:rect.x + rect.w, :], (rect.w // 2, rect.h // 2))
```
Comment (foookinaaa) [inline src/app_predictor/app_predictor.py:60]: what is `(rect.w // 2, rect.h // 2)`?

File: src/app_predictor/app_predictor.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    return img
+
+
+def transform(src_filename, desc_filename):
+    img = cv2.imread(src_filename)
+    faces = predict(img)
+    logger.info(f"Found '{len(faces)}' faces")
+    if faces:
+        logger.info(f"Rects: '{faces}'")
+    img = blur(img, faces)
```
Comment (foookinaaa) [inline src/app_predictor/app_predictor.py:71]: move to `if faces`

File: src/app_receiver/app_receiver.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        self._task_creator = task_creator
+
+    def start(self, update: Update, context: CallbackContext) -> None:
+        user = update.effective_user
+        update.message.reply_markdown_v2(
+            fr'Hi {user.mention_markdown_v2()}\!',
+            reply_markup=ForceReply(selective=True),
+        )
+
+    def help_command(self, update: Update, context: CallbackContext) -> None:
```
Comment (foookinaaa) [inline src/app_receiver/app_receiver.py:44]: add info to user about loading pictures

Comment (mrk-andreev) [reply src/app_receiver/app_receiver.py:44]: Fixed

File: src/app_predictor/app_predictor.py (hunk 2 above)
Comment (mrk-andreev) [reply src/app_predictor/app_predictor.py]: Fixed

File: src/app_predictor/app_predictor.py (hunk 1 above)
Comment (mrk-andreev) [reply src/app_predictor/app_predictor.py]: Fixed

File: envs/local/docker-compose-app.yaml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+      context: ../../src/app_receiver
+      dockerfile: Dockerfile
+    environment:
+      REDIS_QUEUE: "queue:in_images"
+      REDIS_HOST: redis
+      REDIS_PORT: 6379
+      TELEGRAM_BOT_TOKEN: "${TELEGRAM_BOT_TOKEN}"
+
+
+  app_predictor:
```
[the remark this answers is not in the document]
Comment (mrk-andreev) [reply envs/local/docker-compose-app.yaml:24]: Fixed

File: envs/local/docker-compose-app.yaml [hunk 5]
```diff
@@ -0,0 +1,31 @@
+version: "3"
+
+services:
+  redis:
+    restart: always
+    image: redis:7.0.5
+    ports:
+      - "6379:6379"
+
+  app_receiver:
```
[the remark this answers is not in the document]
Comment (mrk-andreev) [reply envs/local/docker-compose-app.yaml:10]: Fixed

File: experiments/v0_prototype/detect-faces.ipynb [hunk 6]
```diff
@@ -0,0 +1,103 @@
+{
```
[the remark this answers is not in the document]
Comment (mrk-andreev) [reply experiments/v0_prototype/detect-faces.ipynb:1]: Fixed

**Pull request merged**